checkPages.js

```diff
@@ -55,6 +55,36 @@ module.exports = function(host, options, done) {
     });
   }
 
+  /* Converts http://examplé.com/rosé?rosé=1 to
+   * http://examplé.com/ros%C3%A9?ros%C3%A9=1 so that servers which expect
+   * ASCII only requests can handle it */
+  function fixNonAsciiLink(link) {
+    var parsed = url.parse(link);
+    if (/^[\u0000-\u007f]*$/.test(parsed.path)) {
```

Comment:
The tricky part is that different libraries behave in different ways. For example, cheerio seems to turn a link like `<a href="café café">` to `café%20café`, escaping the whitespace but not the non-ASCII characters. nock seems to simply accept Javascript strings. Some real servers seem to be able to accept UTF-8, and some only work with escaped URLs. I did consider using `uri-js` to convert URLs to IRIs, but it seems like punycode is unnecessary. Since you're interested I'll put some more work in this.